### DISCUSSION TITLE
Consistency across the app ecosystem

### DISCUSSION THREAD

**Discussion by ericpritchett** [small-tech/watson#11] [Ideas]
### The Challenge
In terms of look and feel, application consistency can become fragmented. I find when applications use the same elements in the same place regardless of what app is being used is really beneficial for people and harder, if even slightly, when they're not. This can include the layout of application such as an app with a sidebar, icons like the settings gear cog, sending something outside the application like Export, Print, and Share (e.g. see Calendar), open an item with the proper file dialog, or create a new item. Having these elements in the same place regardless of the app is very nice!

### The Opportunity
When a developer wants to develop an app there's a good baseline of elements that all applications may share. Some of these I've listed above, but a developer will probably already have an idea of what they'll want to do (e.g. Export and Share -> Print, New Item, Search, etc). Using Watson, if a developer could simply click a few checkboxes with the elements they want and they would then be placed in the proper place, done. For example, a developer may want a Settings/Preferences (gear cog) section with some Interface options like Akira:

![akira_preferences_interface](https://user-images.githubusercontent.com/12094707/136682215-29258276-ce35-4abb-a422-960a500b5513.png)

The dialog box above is an example of an element a lot of apps might use, but may be implemented in an inconsistent way if left to the developer. I think it would be nice to have something like a few checkboxes that would include/build these elements for them and thus increasing the likelihood of application consistency across different apps.

### Scenario (from a person using the application)
Eric wants to export a few things in a few different apps. The behavior is the same regardless of the app. Eric clicks on the "Export or Share" icon in the top right section of the app and selects his actions the same way in all three apps.

### Conclusion
The idea is not to create every element, but rather to create a baseline of most common elements that applications use. I hope you consider this idea and I'm really looking forward to Watson. Thanks for building it!

**Comment by aral** [comment]
Hi Eric,

Thank you for this. I’ve also been thinking about this and I agree, it would be hugely useful to have implementations of basic components. It’s not something that will be in the first release but I’m very open to slowly adding functionality along these lines.

(So say, in addition to your example, it should be easy to create a Welcome screen should you application need one.)

In a way, I guess this is why Granite was started and I wonder how much of it should be configuration in Watson and how much should be documentation that points to useful Granite components, etc.

All that said, what I don’t think Watson should become is a visual app configurator or a heavyweight tool. So we should make sure we make each feature added earn its right to be there via a trial by fire :)

Thanks again for sharing your thoughts and I hope you enjoy playing with Watson once it’s a bit more mature.

(I’m building my first app with it currently and so I already have a list of issues for improvements I want to make.)
